rust/core-lib/src/editor.rs

```diff
@@ -431,18 +419,90 @@ impl<W: Write + Send + 'static> Editor<W> {
         }
     }
 
+    /// Common logic for a number of delete methods. For each region in the selection,
+    /// if the selection is a caret, delete the region between the caret and the
+    /// movement applied to the caret, otherwise delete the region.
+    ///
+    /// If `save` is set, save the deleted text into the kill ring.
+    fn delete_by_movement(&mut self, movement: Movement, save: bool) {
+        // We compute deletions as a selection because the merge logic is convenient.
```

Comment:
Yes, totally reasonable. The only hesitation is that it's _just_ the intervals, so horiz and affinity have no role. This is the familiar question of a small number of general types, or types specialized to their actual role. In this case, the value of specializing is marginal, and being able to reuse the merge logic is a win.